src/plugins/subscription-manager.py

```diff
@@ -159,6 +159,17 @@ def warnOrGiveUsageMessage(conduit):
             conduit.info(2, msg)
 
 
+def init_hook(conduit):
+    disable_system_repos = conduit.confBool('main', 'disable_system_repos', default=False)
+
+    if disable_system_repos:
+        repo_storage = conduit.getRepos()
+        for repo in repo_storage.repos.values():
+            if os.path.basename(repo.repofile) != "redhat.repo":
```

Comment:
It isn't necessary to disable repository that is already disabled. You can add here one more condition:

`enabled is True`.